app/models/async_transaction/vet360/base.rb

```diff
@@ -23,6 +31,50 @@ def self.start(user, response)
           transaction_status: response.transaction.status
         )
       end
+
+      # Updates the status and transaction_status with fresh API data
+      # @params user [User] the user whose tx data is being updated
+      # @params service [Vet360::ContactInformation::Service] an initialized vet360 client
+      # @params tx_id [int] the transaction_id
+      # @returns [AsyncTransaction::Vet360::Base]
+      def self.refresh_transaction_status(user, service, tx_id = nil)
+        transaction_record = Base.find_by!(user_uuid: user.uuid, transaction_id: tx_id)
+        # No need for a API request if this tx is already complete
+        return transaction_record if transaction_record.finished?
+        api_response = Base.fetch_transaction(transaction_record, service)
+        transaction_record.status = COMPLETED if FINAL_STATUSES.include? api_response.transaction.status
+        transaction_record.transaction_status = api_response.transaction.status
+        transaction_record.save!
+        transaction_record
+      end
```

Comment:
There is a lot going on in this `refresh_transaction_status` method.  Perhaps break it up into multiple methods, each with a single purpose?